### PR TITLE
Add sum-of-multiples exercise

### PR DESCRIPTION
**Pull request by palisaide**
Adds 'sum-of-multiples' exercise

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -101,15 +102,22 @@
       "strings",
       "control-flow (if-else statements)",
       "control-flow (loops)"
-	]
```
Comment (palisaide) [inline config.json]: Some tabs were introduced here at some point.  I converted them to spaces.

File: exercises/sum-of-multiples/src/example.c [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#include "sum_of_multiples.h"
+#include <stdbool.h>
+
+unsigned int sum_of_multiples_to(const unsigned int *multiples, const size_t number_of_multiples, const int upper_bound)
```
Comment (patricksjackson) [inline exercises/sum-of-multiples/src/example.c]: `upper_bound` and `*multiples` should have the same type. Same goes for `current` in the loop below.

File: exercises/sum-of-multiples/src/example.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#ifndef _SUM_OF_MULTIPLES_H
+#define _SUM_OF_MULTIPLES_H
+
+#include <stddef.h>
+
+unsigned int sum_of_multiples_to(const unsigned int *multiples, const size_t number_of_multiples, const int up_to);
```
Comment (patricksjackson) [inline exercises/sum-of-multiples/src/example.h]: Is there a reason this isn't just named `sum_of_multiples`?

File: exercises/sum-of-multiples/src/example.c [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+         if ((current % multiples[i]) == 0) {
+            is_multiple = true;
+            break;
+         }
+      }
+
+      if (is_multiple) {
+         sum += current;
+      }
+   }
```
Comment (ryanplusplus) [inline exercises/sum-of-multiples/src/example.c]: Instead of using `is_multiple`, could you just move `sum += current` on line 13?

**Comment by ryanplusplus** [comment]
Looks good outside of a few small nitpicks

File: exercises/sum-of-multiples/src/example.c (hunk 4 above)
Comment (palisaide) [inline exercises/sum-of-multiples/src/example.c]: Haha, yep.  Not sure why I didn't see that.

File: exercises/sum-of-multiples/src/example.c (hunk 2 above)
Comment (palisaide) [inline exercises/sum-of-multiples/src/example.c]: Right, I'll make them all unsigned.

File: exercises/sum-of-multiples/src/example.h (hunk 3 above)
Comment (palisaide) [inline exercises/sum-of-multiples/src/example.h]: I thought `sum_of_multples_to` would be more descriptive.  But now that I read it, I seems excessive.  I'll change it to `sum_of_multiples`

**Comment by palisaide** [comment]
Ready for another look!

**Comment by ryanplusplus** [comment]
LGTM (but needs to have master merged)

**Pull request merged**